scipy/optimize/_lsq/least_squares.py

```diff
@@ -939,14 +964,19 @@ def jac_wrapped(x, f):
             else:
                 tr_solver = 'lsmr'
 
+    # Wrap callback function.  If callback is None, callback_wrapped also is None
+    callback_wrapped = _wrap_callback(callback)
+
     if method == 'lm':
+        if callback is not None:
+            warn("Callback function specified, but not supported with `lm` method.")
```

Comment:
note that this line isn't hit by the testsuite on this branch when I check locally with `pytest-cov`